wallet/chain/p/builder/builder.go

```diff
@@ -324,7 +328,26 @@ func (b *builder) NewBaseTx(
 	toStake := map[ids.ID]uint64{}
 
 	ops := common.NewOptions(options)
-	inputs, changeOutputs, _, err := b.spend(toBurn, toStake, ops)
+	memo := ops.Memo()
+	memoComplexity := feecomponent.Dimensions{
```

Comment:
The builder needs to account for accrued complexity while building the transaction so that it can guarantee that sufficient fees are paid.

So, this complexity calculation is being performed here rather than using `fee.TxComplexity` because the tx hasn't been created yet (this function is creating the tx). This is why things like `OutputComplexity` ect are exported.